src/justfile.rs

```diff
@@ -509,6 +511,27 @@ mod tests {
     }
   }
 
+  #[cfg(target_os = "windows")]
```

Comment:
This change doesn't seem related to this PR.

I think this PR should get a new test which just tests the feature being added.

This could be as simple as:

```rust
#[test]
fn windows_poweshell_setting_uses_powershell() {
  Test::new()
    .justfile(
     "
      set windows-powershell

      foo:
        Write-Output bar
    "#,
    )
    .stdout("bar")
    .stderr("Write-Output bar")
    .run();
}
```

This uses the new, struct-based tests instead of macros, which are preferred at this point, although both kinds of tests still exist.

This test can go in a new module `tests/windows_powershell.rs`